subsys/bluetooth/shell/bt.c

```diff
@@ -1048,8 +1048,8 @@ static int cmd_scan_filter_set_addr(const struct shell *sh, size_t argc,
 {
 	const char *addr_arg = argv[1];
 
-	/* Validate length */
-	if (strlen(addr_arg) > sizeof(scan_filter.addr)) {
+	/* Validate length including null terminator. */
+	if ((strlen(addr_arg) + 1) != sizeof(scan_filter.addr)) {
```

Comment:
```suggestion
	if ((strlen(addr_arg) >= sizeof(scan_filter.addr)) {
```

It should not be `!=` as the filter supports partial address, e.g. you can use `aa:bb:cc` as your address filter